pkg/remote/remote.go

```diff
@@ -126,6 +141,29 @@ func Referrers(ctx context.Context, d name.Digest, option types.RegistryOptions)
 	return nil, errs
 }
 
+func registryMirrors(hostRef name.Reference, option types.RegistryOptions) []name.Reference {
+	var mirrors []name.Reference
+
+	ctx := hostRef.Context()
+	reg := ctx.RegistryStr()
+	if ms, ok := option.RegistryMirrors[reg]; ok {
+		for _, m := range ms {
+			var nameOpts []name.Option
+			if option.Insecure {
+				nameOpts = append(nameOpts, name.Insecure)
+			}
+			mirrorImageName := strings.Replace(hostRef.Name(), reg, m, 1)
+			ref, err := name.ParseReference(mirrorImageName, nameOpts...)
+			if err != nil {
+				log.WithPrefix("remote").Warn("Unable to parse mirror of image", log.String("mirror", mirrorImageName))
```

Comment:
This means users specified the wrong registry. I think we should return an error. What do you think?

Reply:
It make sense. Thanks!
Updated in https://github.com/aquasecurity/trivy/pull/8244/commits/8d269d2f412c4a55cda3ccbd8e4f6efb57867d72